include/wx/gtk/private/glibptr.h

```diff
@@ -0,0 +1,57 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/gtk/private/glibptr.h
+// Purpose:     Simple RAII helper for calling g_free() automatically
+// Author:      Vadim Zeitlin
+// Created:     2024-12-28
+// Copyright:   (c) 2024 Vadim Zeitlin <vadim@wxwidgets.org>
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_GTK_PRIVATE_OWNEDPTR_H_
+#define _WX_GTK_PRIVATE_OWNEDPTR_H_
+
+// ----------------------------------------------------------------------------
+// Convenience class for calling g_free() automatically
+// ----------------------------------------------------------------------------
+
+template <typename T>
+class wxGlibPtr
+{
+public:
+    wxGlibPtr() = default;
+    explicit wxGlibPtr(const T *p) : m_ptr(const_cast<T*>(p)) { }
+    wxGlibPtr(wxGlibPtr&& other) : m_ptr(other.m_ptr) { other.m_ptr = nullptr; }
+    ~wxGlibPtr() { g_free(m_ptr); }
+
+    wxGlibPtr& operator=(wxGlibPtr&& other)
+    {
+        if ( &other != this )
+        {
+            g_free(m_ptr);
+            m_ptr = other.m_ptr;
+            other.m_ptr = nullptr;
+        }
+
+        return *this;
+    }
```

Comment:
We could avoid branching here by swapping the pointers. The moved-from object will be destroyed eventually:

```suggestion
    wxGlibPtr& operator=(wxGlibPtr&& other)
    {
        std::swap(m_ptr, other.m_ptr);
        return *this;
    }
```